### ISSUE TITLE
`intersection` gives incorrect result with generators

### ISSUE THREAD

**Issue by airikh** [dgilland/fnc#2]
Hi Derrick,

I recently noticed that `intersection` behaves incorrectly when a generator is provided in `*seqs`. 

For example: 
```python
>>> [*fnc.intersection(
...     [2, 1], 
...     (x for x in [2, 1])
... )]
[2]
```
It seems the issue arrises from iterating each iterable in `*seqs` once per element in `seq`, because any generator in `*seqs` is exhausted after the first element of `seq`. 

The same behavior is present in v0.3.0 and v0.4.0.

If you want me to take a stab at an improvement, let me know. I couldn't think of a good approach off the top of my head, but you're brilliant at this stuff, so I figured I'd run it by you first.

Hope all is well!

**Comment by dgilland** [comment]
Hey Eric,

Sure, take a crack at it. It's a known limitation that it doesn't work with generators as inputs (except as the first argument). Will be curious to see what you come up with!

**Comment by dgilland** [comment]
@airikh FWIW, I've fixed this issue in 80f640b and will release a bugfix version some time this week.
